File: XML2LFES/XML2LFES.m
function [myLFES,S]=XML2LFES(xmlFile)
%%  Copyright 2018 Engineering Systems Analytics LLC
%
% This function converts an XML file into an LFES structure (version 2).
% Proposed Usage:  LFES=XML2LFES(XMLFile)
disp(['I am entering ' 'XML2LFES.m'])
S=xml2struct(xmlFile);

%% Initialize the LFES Structure
myLFES = initializeLFES;

%% Fill the High LFES Attributes
myLFES=insertObjB(myLFES,S.LFES.Attributes);

%% Setup Machines
if ismember('Machine',fieldnames(S.LFES))
    myLFES = setupMachines(myLFES, S);
else
    myLFES.machines.names=cell(0);
end

%% Setup Independent Buffers
if ismember('IndBuffer',fieldnames(S.LFES))
    myLFES = setupIndBuffers(myLFES, S);
else
    myLFES.indBuffers.names=cell(0);
end

%% Setup Transporters
if ismember('Transporter',fieldnames(S.LFES))
    myLFES = setupTransporters(myLFES, S);
else
    myLFES.transporters.names=cell(0);
end

%% Setup Controllers
if ismember('Controller',fieldnames(S.LFES))
    myLFES = setupControllers(myLFES, S);
else
    myLFES.controllers.names=cell(0);
end

%% Setup Services
if ismember('Service',fieldnames(S.LFES))
    myLFES = setupServices(myLFES, S);
else
    myLFES.services.names=cell(0);
end

%% Setup Abstract Transportation Methods
if ismember('Abstractions',fieldnames(S.LFES))
    myLFES = setupAbstractTransportationMethods(myLFES, S);
end

%% Setup Transportation Process Abstractions
disp(['I am leaving  ' 'XML2LFES.m']);

File: XML2LFES/XML2LFES.py
"""
Copyright 2020 LIINES
@company: Thayer School of Engineering at Dartmouth
@lab: LIINES Lab
@Modified: 01/20/2022

"""

from .XML2LFES_Functions import *

def XML2LFES(xmlfile, verboseMode):
    '''
    This function reads the XML input file, instantiates a LFES called myLFES,
    and populates the instance with its constituents, along with their attributes and methods.
    :param xmlFile: Input file representing the system
    :return: myLFES: An instantiated Large Flexible Engineering System object ready to compute HFGT mathematical models
    '''

    print("I am entering XML2LFES.py")
    S = XML2Struct(xmlfile)
    
    # Initialize the LFES Structure
    myLFES = LFES()
    
    # Fill the High LFES Attributes
    insertAttributes(myLFES, S[1].attrib)

    if verboseMode <= 2:
        # Setup Machines
        if 'Machine' in S[2]:
            setupMachines(myLFES, S[1])
        else:
            myLFES.machines.name = []

        # Setup Independent Buffers
        if 'IndBuffer'in S[2]:
            setupIndBuffers(myLFES, S[1])
        else:
            myLFES.indBuffers.name = []

        # Setup Transporters
        if 'Transporter' in S[2]:
            setupTransporters(myLFES, S[1])
        else:
            myLFES.transporters.name = []
    else:
        setupMethodIndicies(myLFES, S[1])
        
    # Setup Services
    if 'Service' in S[2]:
        setupServices(myLFES, S[1])
    else:
        myLFES.services.name = []
        
    # Setup Controllers
    if 'Controller' in S[2]:
        setupControllers(myLFES, S[1])
    else:
        myLFES.controllers.name = []
        
    # Setup Operands
    if 'Operand' in S[2]:
        setupOperands(myLFES, S[1])
    else:
        myLFES.setOperands = []

    # Setup Abstract Transportation Methods
    if 'Abstractions' in S[2]:
        setupAbstractMethods(myLFES, S[1])

    print("I am exiting XML2LFES.py")
    return myLFES
